clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,7 +496,14 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
```

Comment:
The downside of any of the suggested approaches is that it won't help already released versions. Another option would be to use an existing RetriableException when dealing with older clients. This would require bumping the relevant protocol versions though.

Going forward, it may make sense to define a `RetriableUnknownException` that the broker can use to force older clients to retry.